project_task_code/models/project_task.py

```diff
@@ -42,3 +42,15 @@ def name_get(self):
             name = "[{}] {}".format(rec.code, task[1])
             new_result.append((rec.id, name))
         return new_result
+
+    @api.model
+    def _name_search(
+        self, name, args=None, operator="ilike", limit=100, name_get_uid=None
+    ):
+        if not args:
+            args = []
+        if name:
+            args += ["|", ["code", "ilike", name]]
```

Comment:
I would suggest using `expression.AND` whenever you need to aggregate two domains. You can  an example [here](https://search.odooism.com/?q=expression.AND&i=nope&files=&excludeFiles=po%24%7Cpot%24%7Cyml%24%7Cyaml%24%7Ccss%24%7C%2Fstatic%2Flib%2F%7C%2Ftests%2F&repos=15.0)

Reply:
@ivs-cetmix did you mean `expression.OR`? It seems like in our case this one is more suitable

Reply:
> @ivs-cetmix did you mean `expression.OR`? It seems like in our case [this one](https://search.odooism.com/?q=expression.OR&i=nope&files=&excludeFiles=po%24%7Cpot%24%7Cyml%24%7Cyaml%24%7Ccss%24%7C%2Fstatic%2Flib%2F%7C%2Ftests%2F&repos=15.0) is more suitable

Yes, you are right. My mistake)